core/shared/src/main/scala/zio/ConfigProvider.scala

```diff
@@ -209,7 +209,7 @@ object ConfigProvider {
                         case (Left(e1), Left(e2)) => ZIO.fail(e1 && e2)
                         case (Left(_), Right(r))  => ZIO.succeed(r)
                         case (Right(l), Left(_))  => ZIO.succeed(l)
-                        case (Right(l), Right(r)) => ZIO.succeed(l ++ r)
+                        case (Right(l), Right(r)) => ZIO.succeed(if (l.nonEmpty) l else r)
```

Comment:
@adamgfraser I think there might be another bug here. In some cases, a failure on the left side is represented as a `Right(Set())`. Is this the intended behaviour?

Reply:
@kyri-petrou Will take a look at clarifying the contract around that!